src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2153,6 +2153,25 @@ impl Coordinator {
             typ,
         );
 
+        // We have the opportunity to name an `until` frontier that will prevent work we needn't perform.
+        // By default, `until` will be `Antichain::new()`, which prevents no updates and is safe.
+        // We do this here so that `optimize_dataflow` can pick up the monotonicity information.
+        if let Some(as_of) = dataflow.as_of.as_ref() {
+            if !as_of.is_empty() {
+                if let Some(next) = as_of.elements()[0].checked_add(1) {
```

Comment:
```suggestion
                if let Some(next) = as_of.into_option().and_then(|as_of| as_of.checked_add(1)) {

```

Reply:
The compiler complains here that `*as_of` cannot be moved. We can do instead:
```suggestion
                if let Some(next) = as_of
                    .clone()
                    .into_option()
                    .and_then(|as_of| as_of.checked_add(1))
                {
```
Not sure if adding the `clone()` call there gets things to be more confusing than the previous formulation? What do you think would be more idiomatic?

Reply:
You can do `as_option` instead?

Reply:
Yep, done; thanks!